hipe4ml/model_handler.py

```diff
@@ -404,7 +404,22 @@ def __objective(trial):
             study = optuna.create_study(
                 direction=direction, sampler=optuna_sampler)
 
-        study.optimize(__objective, **kwargs)
+        def __callbacks(study, trial):
+
+            if direction == 'maximize':
+                if (study.best_trial.value - trial.value > early_stopping) and \
+                    (np.abs(trial.number - study.best_trial.number > 10)):
+                    study.stop()
+            if direction == 'minimize':
+                if (trial.value - study.best_trial.value > early_stopping) and \
+                    (np.abs(trial.number - study.best_trial.number > 10)):
+                    study.stop()
```

Comment:
What's the idea here to define when to stop the optimisation? From the example plot you showed, the optimisation could be wrongly stopped too early.